Todo.WebApi/Controllers/TodoController.cs

```diff
@@ -33,9 +33,29 @@ public IActionResult ListAll()
         public IActionResult Add([FromBody]TodoTask task)
         {
             var list = _todoLists.GetPublic();
-            if(list is null) { return NotFound("The public list doesn't exist."); }
+            if (list is null) { return NotFound("The public list doesn't exist."); }
+
+            if (task.Name != null && task.Name.Length > 40)
+            {
+                return BadRequest("Name can be no longer than 40 characters.");
+            }
+
+            if (task.Name is null)
+            {
+                if (task.Description.Length < 40)
+                {
+                    task.Name = task.Description;
+                    task.Description = string.Empty;
+                }
+                else
+                {
+                    var descriptionSnip = task.Description.Substring(0, 37);
+                    task.Name = $"{descriptionSnip}�";
+                }
```

Comment:
We can extract this snippet into a new method. It could be something like:
```cs
if(task.Name is null)
{
    EnsureTaskHasName(task);
}
```
This method can directly modify the `task` and since it's a reference, it will be modified for the rest of this code. 🙂